keys.go

```diff
@@ -133,103 +133,96 @@ type KeyVersion struct {
 	CreationDate *time.Time `json:"creationDate,omitempty"`
 }
 
-// CreateKey creates a new KP key.
-func (c *Client) CreateKey(ctx context.Context, name string, expiration *time.Time, extractable bool) (*Key, error) {
-	return c.CreateImportedKey(ctx, name, expiration, "", "", "", extractable)
-}
-
-// CreateImportedKey creates a new KP key from the given key material.
-func (c *Client) CreateImportedKey(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string, extractable bool) (*Key, error) {
-	key := c.createKeyTemplate(ctx, name, expiration, payload, encryptedNonce, iv, extractable, nil, AlgorithmRSAOAEP256, nil)
-	return c.createKey(ctx, key)
-}
+// This function returns a string so we can pass extra info not in the key structu if needed
+type CreateKeyOption func(k *Key) string
 
-// CreateImportedKeyWithSHA1 creates a new KP key from the given key material
-// using RSAES OAEP SHA 1 as encryption algorithm.
-func (c *Client) CreateImportedKeyWithSHA1(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string, extractable bool, aliases []string) (*Key, error) {
-	key := c.createKeyTemplate(ctx, name, expiration, payload, encryptedNonce, iv, extractable, aliases, AlgorithmRSAOAEP1, nil)
-	return c.createKey(ctx, key)
-}
-
-// CreateRootKey creates a new, non-extractable key resource without
-// key material.
-func (c *Client) CreateRootKey(ctx context.Context, name string, expiration *time.Time) (*Key, error) {
-	return c.CreateKey(ctx, name, expiration, false)
+func WithExpiration(expiration *time.Time) CreateKeyOption {
+	return func(key *Key) string {
+		key.Expiration = expiration
+		return "expiration"
+	}
 }
 
-// CreateStandardKey creates a new, extractable key resource without
-// key material.
-func (c *Client) CreateStandardKey(ctx context.Context, name string, expiration *time.Time) (*Key, error) {
-	return c.CreateKey(ctx, name, expiration, true)
+func WithDescription(description string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Description = description
+		return "description"
+	}
 }
 
-// CreateImportedRootKey creates a new, non-extractable key resource
-// with the given key material.
-func (c *Client) CreateImportedRootKey(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string) (*Key, error) {
-	return c.CreateImportedKey(ctx, name, expiration, payload, encryptedNonce, iv, false)
+func WithPayload(payload, encryptedNonce, iv string, sha1 bool) CreateKeyOption {
```

Comment:
The one thing I feel like i could improve even better is that for Standard Keys, the nonce and iv are not used/needed. However, the original code did have a set of conditionals for checking if the key is extractable or not before setting the nonce and iv, so i preserved that behaviour, and figured i didn't want to allow a `WithStandardKeyPayload` function.